File: Source/Extensions/Tessa.Extensions.Default.Client/UI/KrProcess/StageHandlers/AddFromTemplateUIHandler.cs
#nullable enable

using System;
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.Platform.Storage;
using Tessa.Platform.Validation;

namespace Tessa.Extensions.Default.Client.UI.KrProcess.StageHandlers
{
    /// <summary>
    /// UI обработчик типа этапа <see cref="StageTypeDescriptors.AddFromTemplateDescriptor"/>.
    /// </summary>
    public class AddFromTemplateUIHandler :
        StageTypeUIHandlerBase
    {
        #region Base Overrides

        /// <inheritdoc/>
        public override Task Validate(IKrStageTypeUIHandlerContext context)
        {
            var template = context.Row.TryGet<Guid?>(KrConstants.KrAddFromTemplateSettingsVirtual.FileTemplateID);
            if (!template.HasValue)
            {
                context.ValidationResult.AddWarning(this, "$KrStages_AddFromTemplate_TemplateIsRequiredWarning");
            }

            return Task.CompletedTask;
        }

        #endregion
    }
}

File: WebClient SDK/src/default/ui/krProcess/stageHandlers/addFromTemplateUIHandler.ts
import {
  IKrStageTypeUIHandlerContext,
  KrStageTypeUIHandler,
  StageTypeHandlerDescriptor,
  addFromTemplateDescriptor
} from 'tessa/workflow/krProcess';
import { ValidationResult, ValidationResultType } from 'tessa/platform/validation';

import { plainColumnName } from 'tessa/workflow';

/**
 * UI обработчик типа этапа {@link addFromTemplateDescriptor}.
 */
export class AddFromTemplateUIHandler extends KrStageTypeUIHandler {
  //#endregion fields

  private static readonly _fileTemplateIdFieldName: string = plainColumnName(
    'KrAddFromTemplateSettingsVirtual',
    'FileTemplateID'
  );

  //#endregion

  //#region base overrides

  public descriptors(): StageTypeHandlerDescriptor[] {
    return [addFromTemplateDescriptor];
  }

  public async validate(context: IKrStageTypeUIHandlerContext): Promise<void> {
    if (!context.row.tryGet(AddFromTemplateUIHandler._fileTemplateIdFieldName)) {
      context.validationResult.add(
        ValidationResult.fromText(
          '$KrStages_AddFromTemplate_TemplateIsRequiredWarning',
          ValidationResultType.Warning
        )
      );
    }
  }

  //#endregion
}
